### ISSUE TITLE
Add other Halstead metrics

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#1883]
We are using N1, n1, N2 and n2, but others are available in rust-code-analysis now.

```
            |- halstead
            |  |- n1: 6
            |  |- N1: 86
            |  |- n2: 69
            |  |- N2: 147
            |  |- length: 233
            |  |- estimated program length: 436.9979625220206
            |  |- purity ratio: 1.8755277361460112
            |  |- vocabulary: 75
            |  |- volume: 1451.3147548855402
            |  |- difficulty: 6.391304347826087
            |  |- level: 0.1564625850340136
            |  |- effort: 9275.794302964105
            |  |- time: 515.3219057202281
            |  `- bugs: 0.14715651181813164
```

**Comment by sayantan1413** [comment]
Hi, I am new to this contribution and I would like to work on this issue so can you help me to start with it? @marco-c

**Comment by devkumar24** [comment]
hey, I want to work on the issue, can you assign me for this issue?

**Comment by marco-c** [comment]
@devkumar24 you can work on any issue which has no PR linked to it, no need to be assigned ;)

**Comment by Rishabhc711** [comment]
Hey! Where exactly does the code lie for the computed metrics? How do we look for it?

**Comment by marco-c** [comment]
@Rishabhc711 in bugbug/repository.py and in bugbug/commit_features.py (e.g. search for `halstead_N1` in those files).

**Comment by Amotul-raheem** [comment]
Hi, I'm one of outreachy's participants. I want to work on this issue.

**Comment by suhaibmujahid** [comment]
Resolved by #2372
